### PR TITLE
Setup Smart Screen & Start New stake flow

### PR DESCRIPTION
**Pull request by NovacAndrei**

### CODE REVIEW

File: src/components/widgets/components/amount-input/amount-input.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                        testID="enter-amount"
+                        style={styles.inputText}
+                        autoCapitalize={'none'}
+                        autoCorrect={false}
+                        selectionColor={theme.colors.accent}
+                        value={inputAmount}
+                        onChangeText={text => {
+                            text = text.replace(/,/g, '.');
+                            this.props.setScreenInputData(flowId, text, 'inputAmount');
+                            this.props.actions.runScreenValidation(
```
Comment (krisboit) [inline src/components/widgets/components/amount-input/amount-input.tsx:102]: you can call this action inside `setScreenInputData`, you have there everything you need.

File: src/redux/ui/screens/input-data/actions.ts [hunk 2]
```diff
@@ -24,12 +33,12 @@ export const toggleValidatorMultiple = (
     });
 };
 
-export const selectInput = (screenKey: string, inputData: any, inputKey: string) => async (
+export const setScreenInputData = (screenKey: string, inputData: any, inputKey: string) => async (
```
Comment (krisboit) [inline src/redux/ui/screens/input-data/actions.ts:36]: in some cases we may need to add more fields at once, you can merge `inputData: any, inputKey: string` into an object:
```ts
{
  [key: string]: any
}
```

File: src/redux/ui/screens/input-data/state.ts [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
+    // Save here any input data
+    data: {
+        [key: string]: any;
     };
 }
 
-export interface IScreenInputState {
-    [screen: string]: IScreenInputData;
+export interface IScreenInputDataValidations {
+    fieldsErrors: {
```
Comment (krisboit) [inline src/redux/ui/screens/input-data/state.ts:16]: something is missing here :) there is no info about the field name
Transform this into:
```ts
fieldsErrors: {
  [fieldName: string]: {
    type: 'ERROR_MSG' | 'WARN_MSG';
    message: string;
  }[];
}

File: src/components/widgets/components/amount-input/amount-input.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                    <TextInput
+                        testID="enter-amount"
+                        style={styles.inputText}
+                        autoCapitalize={'none'}
+                        autoCorrect={false}
+                        selectionColor={theme.colors.accent}
+                        value={inputAmount}
+                        onChangeText={text => {
+                            text = text.replace(/,/g, '.');
+                            this.props.setScreenInputData(flowId, text, 'inputAmount');
```
Comment (krisboit) [inline src/components/widgets/components/amount-input/amount-input.tsx:101]: we need to talk a bit about this, I think you misunderstood something

Comment (krisboit) [reply src/components/widgets/components/amount-input/amount-input.tsx:101]: the component is fine, we just need to refine the data flow :)

**Review by georgeburduhos — APPROVED** [review]

File: src/redux/ui/screens/data/actions.ts [hunk 5]
```diff
@@ -223,36 +235,202 @@ export const handleCta = (
...
+                                token,
+                                undefined, // feeOptions
+                                undefined
+                            )(dispatch, getState);
+                        }
+                    }
+                    break;
+                }
+
+                case 'switchNodeDelegateToValidator': {
```
Comment (krisboit) [inline src/redux/ui/screens/data/actions.ts:283]: Let's keep consistent naming for actions, use either camel case of uppercase with underline (e.g. SINGLE_SELECTION). I'm fine either way, but keep them all in the same pattern

File: src/redux/ui/screens/input-data/validation/min-amount-to-stake.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    const tokenConfig = getTokenConfig(blockchain, token.symbol);
+
+    const inputAmount = inputData?.amount;
+    const inputAmountToStd = blockchainInstance.account.amountToStd(
+        new BigNumber(inputAmount),
+        tokenConfig.decimals
+    );
+
+    // TODO
```
Comment (krisboit) [inline src/redux/ui/screens/input-data/validation/min-amount-to-stake.ts:41]: we could send this value from api, inside a details field, so things would get easier.

**Review by krisboit — APPROVED** [review]

**Pull request merged**